hummingbot/market/binance/binance_market.pyx

```diff
@@ -972,15 +972,18 @@ cdef class BinanceMarket(MarketBase):
             self.c_start_tracking_order(order_id, -1, symbol, True, decimal_amount, order_type)
             order_result = None
             if order_type is OrderType.LIMIT:
+                order_price = f"{Decimal(str(price)):f}"
```

Comment:
This is no good. You should quantize the price first, and then convert to decimal string.

Reply:
order price is already quantized, but the problem is if you do Decimal(0.002485), its converted to 0.002485000000999 or some garbage value

Reply:
So first convert it to string so that you can use the format :f for decimal

Reply:
order price is not quantized. See the type definition of c_buy() and c_sell() in `MarketBase`. It is intentionally designed that way to offload the order quantization work from strategy code.

Reply:
but let me try that method as well

Reply:
And by quantize I'm meaning you need to go call `c_quantize_order_price()` as I've commented before. Just converting a floating point to random decimal precision is NOT 'quantizing'. You need to consider the exchange's rules about that particular trading symbol's allowed precision for price and order sizes. That's why that function exists. Don't just go `Decimal(str(price))` and tell me that's quantization.

Reply:
The Pure MM strategy already quantizes it before sending it to binance market, which is what I meant not the above

Reply:
I will quantize it again in the binance market class

Reply:
The market classes is supposed to work for all strategies, how the pure MM handles it is irrelevant. In fact... if you pass in a price or order size that's not a floating point then that's wrong because you're violating the argument type specs.

Reply:
I have added the quantization. However just using the :f tag to the string drops precision from 8 to 6 digits.

Reply:
So I have kept the original method. Let me know if you want to implement it in some other way.

Reply:
Oh so basically this is from the bug yingdan posted for iost token, so I am testing it there

Reply:
This is not a reason to keep the original method. You should find out why the quantization gave you the 6 digits price, and find out whether that's correct or not by comparing it to Binance's quantization rules from their PRICE_FILTER spec. We've tested today that the :f tag can give you >24 digits so that's not the cause of it.